src/client.ts

```diff
@@ -3,8 +3,16 @@ import * as assert from 'assert';
 import safe from '@smartlyio/safe-navigation';
 
 type HeaderProp<H, Next> = H extends void ? Next : { headers: H } & Next;
-type QueryProp<Q, Next> = Q extends void ? Next : { query: Q } & Next;
-type BodyProp<B> = B extends void ? object : { body: B };
+type QueryProp<Q, Next> = true extends HasOnlyOptionalTypes<Q>
```

Comment:
basically try to convince the ts compiler to allow `.get()`, `.get({query: nn})`, and `.get({query:nn, otherstuff })` when all the query parameters are optional

Reply:
Those types got really nested and hard to follow. Do you think something like this is more straight forward to read? It should (hopefully) behave the same

```ts
type NotVoidKeys<T> = {
  [P in keyof T]: T[P] extends void ? never : P;
}[keyof T];
type RemoveVoidProps<T> = Pick<T, NotVoidKeys<T>>;

type HasOnlyOptionalTypes<O> = Partial<O> extends O ? true : false;

export type ClientArg<
  H extends server.Headers | void,
  Q extends server.Query | void,
  B extends server.RequestBody<any> | void
> = RemoveVoidProps<
  {
    headers: H;
    body: B;
  } & (true extends HasOnlyOptionalTypes<Q> ? { query?: Q } : { query: Q })
>;
```